task/clair-scan/0.1/clair-scan.yaml

```diff
@@ -24,7 +24,39 @@ spec:
       description: Tekton task test output.
     - name: CLAIR_SCAN_RESULT
       description: Clair scan result.
+    - name: IMAGES_PROCESSED
+      description: Images processed in the task.
   steps:
+    - name: get-image-manifests
+      image: quay.io/redhat-appstudio/hacbs-test:v1.3.2@sha256:565dd98e46ca9f011dee8b700600839dc1f76d1e66b065af1cb77c97a7ecee15
+      env:
+        - name: IMAGE_URL
+          value: $(params.image-url)
+        - name: IMAGE_DIGEST
+          value: $(params.image-digest)
+      securityContext:
+        capabilities:
+          add:
+            - SETFCAP
+      script: |
+        #!/usr/bin/env bash
+        set -euo pipefail
+        . /utils.sh
+
+        imagewithouttag=$(echo $IMAGE_URL | sed "s/\(.*\):.*/\1/" | tr -d '\n')
+        # strip new-line escape symbol from parameter and save it to variable
+        imageanddigest=$(echo $imagewithouttag@$IMAGE_DIGEST)
+        echo "Inspecting raw image manifest $imageanddigest."
+
+        # check if the image is a manifest list
+        image_index_manifests=$(get_image_index_manifests -i ${imageanddigest})
+        if [ -n "$image_index_manifests" ]; then
+          echo "$image_index_manifests" | jq -r 'to_entries[] | "\(.key) \(.value)"' | while read -r arch arch_sha; do
+            echo "$arch_sha" > /tekton/home/image-manifest-$arch.sha
```

Comment:
What prevents us from merging these steps and usie JSON directly? Every extra step causes extra load, because tekton runs all step containers in paralel.

Reply:
@MartinBasti So the original file has two steps, one using the clair image and the other using the hacbs-test image. The issue is that the clair image doesn't have `jq` or `skopeo` so I cannot use that step and we also needed to inspect the image before the clair step. Hence, I had to create a third step.

Reply:
okay, it may deserve comment in code, so I won't find in the hard way during refactoring in future

Reply:
Done :)